src/assets/scripts/aircraft/AircraftInstanceModel.js

```diff
@@ -1705,6 +1705,41 @@ export default class Aircraft {
     }
 
     /**
+     * Checks if the aircraft is inside the airspace of a specified airport
+     *
+     * @for AircraftInstanceModel
+     * @method isInsideAirspace
+     * @param  {airport} airport the airport whose airspace we are checking
+     * @return {Boolean}
+     * @private
+     */
+    isInsideAirspace(airport) {
+        let withinAirspaceLateralBoundaries = this.distance <= airport.ctr_radius;
+        const withinAirspaceAltitudeRange = this.altitude <= airport.ctr_ceiling;
+
+        if (!_isNil(airport.perimeter)) {    // polygonal airspace boundary
+            withinAirspaceLateralBoundaries = point_in_area(this.position, airport.perimeter);
+        }
+
+        return withinAirspaceAltitudeRange && withinAirspaceLateralBoundaries;
+    }
+
+    /**
+     * Aircraft has "weight-on-wheels" (on the ground)
+     * @for AircraftInstanceModel
+     * @method isOnGround
+     */
+    isOnGround() {
+        const error_allowance_ft = 5;
+        const airport = window.airportController.airport_get();
+        const runway = airport.getRunway(this.rwy_dep || this.rwy_arr);
```

Comment:
yeah, I understand.  and you're right, the getter would be in the `AircraftInstanceModel`.

Essentially whats happening here, from what I understand, is 'give me the runway you used/or are going to use'.  This is where the getter would be helpful because it can do that logic on its own.  Thus, anything calling the getter simply says 'give me the runway info' and the getter figures out which one it should send back.  and then everything is right in the world.

But, if this isn't used all over the place, it probably isn't really worth the effort.